fixture-monkey-jakarta-validation/src/main/java/com/navercorp/fixturemonkey/jakarta/validation/introspector/JakartaValidationConstraintGenerator.java

```diff
@@ -292,180 +292,158 @@ public JavaIntegerConstraint generateIntegerConstraint(ArbitraryGeneratorContext
+				min = BigDecimal.ZERO;
+				minInclusive = false;
 			}
+		}
 
-			if (positiveMax == null) {
-				positiveMax = decimalMax;
-			} else if (positiveMax.compareTo(decimalMax) > 0) {
-				positiveMax = decimalMax;
+		if (context.findAnnotation(PositiveOrZero.class).isPresent()) {
+			if (min == null || BigDecimal.ZERO.compareTo(min) > 0) {
+				min = BigDecimal.ZERO;
+				minInclusive = true;
 			}
 		}
 
 		if (context.findAnnotation(Negative.class).isPresent()) {
-			if (negativeMax == null || negativeMax.compareTo(BigDecimal.ZERO) > 0) {
-				negativeMax = BigDecimal.ZERO;
-				negativeMaxInclusive = false;
+			if (max == null || BigDecimal.ZERO.compareTo(max) < 0
+				|| (BigDecimal.ZERO.compareTo(max) == 0 && maxInclusive)) {
+				max = BigDecimal.ZERO;
+				maxInclusive = false;
 			}
 		}
 
 		if (context.findAnnotation(NegativeOrZero.class).isPresent()) {
-			if (negativeMax == null || negativeMax.compareTo(BigDecimal.ZERO) > 0) {
-				negativeMax = BigDecimal.ZERO;
-				negativeMaxInclusive = true;
+			if (max == null || BigDecimal.ZERO.compareTo(max) < 0) {
+				max = BigDecimal.ZERO;
+				maxInclusive = true;
 			}
 		}
 
-		if (context.findAnnotation(Positive.class).isPresent()) {
-			if (positiveMin == null || positiveMin.compareTo(BigDecimal.ZERO) < 0) {
-				positiveMin = BigDecimal.ZERO;
-				positiveMinInclusive = false;
+		Optional<Digits> digitsAnn = context.findAnnotation(Digits.class);
+		if (digitsAnn.isPresent()) {
+			Digits digits = digitsAnn.get();
+			int integerDigits = digits.integer();
+			int fractionDigits = digits.fraction();
+
+			StringBuilder maxBuilder = new StringBuilder();
+			for (int i = 0; i < integerDigits; i++) {
+				maxBuilder.append('9');
 			}
-		}
+			if (fractionDigits > 0) {
+				maxBuilder.append('.');
+				for (int i = 0; i < fractionDigits; i++) {
+					maxBuilder.append('9');
+				}
+			}
+			BigDecimal digitsMax = new BigDecimal(maxBuilder.toString());
+			BigDecimal digitsMin = digitsMax.negate();
 
-		if (context.findAnnotation(PositiveOrZero.class).isPresent()) {
-			if (positiveMin == null || positiveMin.compareTo(BigDecimal.ZERO) < 0) {
-				positiveMin = BigDecimal.ZERO;
-				positiveMinInclusive = true;
+			if (max == null || digitsMax.compareTo(max) < 0) {
+				max = digitsMax;
+				maxInclusive = true;
+			}
+			if (min == null || digitsMin.compareTo(min) > 0) {
+				min = digitsMin;
+				minInclusive = true;
 			}
+
+			scale = digits.fraction();
 		}
 
-		if (positiveMin == null && positiveMax == null && negativeMin == null && negativeMax == null && scale == null) {
+		if (min == null && max == null) {
 			return null;
 		}
 
+		boolean isPositiveMin = min != null && min.compareTo(BigDecimal.ZERO) >= 0;
+		boolean isPositiveMax = max != null && max.compareTo(BigDecimal.ZERO) >= 0;
+		boolean isNegativeMin = min != null && min.compareTo(BigDecimal.ZERO) < 0;
+		boolean isNegativeMax = max != null && max.compareTo(BigDecimal.ZERO) < 0;
```

Comment:
NIT, in convention of FixtureMonkey, only the boolean parameter allows the prefix of `is`.